Package.swift

```diff
@@ -1,47 +1,35 @@
-// swift-tools-version:5.2
+// swift-tools-version:5.3
 import PackageDescription
 
+// Use the appropriate version of SwiftSyntax based on the current compiler
+#if compiler(>=5.5)
+let swiftSyntaxVersion: Package.Dependency.Requirement = .exact("0.50600.1")
```

Comment:
Ok I overlooked that this would only work on macOS when we are using StaticSwiftSyntaxParser. 

I can bring 0.50500.0 back in for Swift 5.5 compiler, I was just trying to use the static linking in Swift 5.5 too so that we could leverage the portability without having to move onto Xcode 13.3 😅 

I'll keep thinking to see if there are cleaner ways to do it